internal/pipeline/pipeline.go

```diff
@@ -60,6 +73,69 @@ type Pipeline struct {
 	Validators []Function `yaml:"validators,omitempty"`
 }
 
+// String returns the string representation of Pipeline struct
+// The string returnes is
+func (p *Pipeline) String() string {
+	return fmt.Sprintf("%+v", *p)
+}
+
+// New returns a pointer to a new default Pipeline.
+// The default Pipeline should be:
+// apiVersion: kpt.dev/v1alpha1
+// kind: Pipeline
+// metadata:
+//   name: pipeline
+// sources:
+//   - '.*'
+func New() *Pipeline {
+	return &Pipeline{
+		ResourceMeta: yaml.ResourceMeta{
+			TypeMeta: yaml.TypeMeta{
+				APIVersion: kptAPIVersion,
+				Kind:       pipelineKind,
+			},
+			ObjectMeta: yaml.ObjectMeta{
+				NameMeta: yaml.NameMeta{
+					Name: defaultName,
+				},
+			},
+		},
+		Sources: defaultSources,
+	}
+}
+
+// fromBytes returns a Pipeline parsed from bytes
+func fromBytes(b []byte) (*Pipeline, error) {
+	p := New()
+	err := yaml.Unmarshal(b, p)
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct pipeline from bytes: %w", err)
+	}
+	return p, nil
+}
+
+// FromReader returns a Pipeline parsed from the content in reader
+func FromReader(r io.Reader) (*Pipeline, error) {
```

Comment:
Do these ctor's perform any validations on the input?

Please document this here.

For example, at which point are you going to validate the syntax. For example, `sources` can only be relative directories? At which point are you going to assert required fields? etc.

Reply:
I will leave a todo here since validating pipeline schema is the next task.